### ISSUE TITLE
.NET 5 build cannot run on OSX 10.15.7

### ISSUE THREAD

**Issue by cousinbenson** [OpenRA/OpenRA#19658]
## Issue Summary

OSX Catalina 10.15.7 cannot run net5 builds due to the following error:
`-[NSOpenGLContext setView:] must be called from the main thread.`

## Reproduction

On OSX 10.15.7:
* Install .NET 5
* Run `make`
* Run `./launch-game.sh Game.Mod=ra`
* Observe crash

## Expected behavior

Game should run.

## Screenshots / Screen recordings / Replays

<img width="1040" alt="Screenshot 2021-09-09 at 14 13 48" src="https://user-images.githubusercontent.com/7032988/132692531-cc7cfc4f-168f-4364-bb9d-641a34f9d89e.png">

**Comment by Mailaender** [comment]
Same problem here https://pastebin.com/AXeYBLvd using a `.dmg` build.

**Comment by pchote** [comment]
I did some digging here and found that this appears to be related to a change in the macOS 10.15 platform SDK that turned a warning into a runtime error. mono is likely built against the old SDK and dotnet against the new one, explaining why the compatibility builds continue to work.

https://github.com/libsdl-org/SDL/commit/309d6137aee5cb406b9371d6bf6c80d8b09a41ca was shipped in SDL 2.0.14 (we curently use 2.0.12) and should fix the issue. I attempted to test the latest SDL (now 2.0.16) in a macOS 10.15.7 virtual machine, but it seems that there were other changes that aren't compatible with the emulated software GL implementation in the virtual machine - I expect this won't be a problem with real hardware.
